daemon/list.go

```diff
@@ -594,11 +594,16 @@ func (daemon *Daemon) refreshImage(s *container.Snapshot, ctx *listContext) (*ty
 	return &c, nil
 }
 
-func populateImageFilterByParents(ancestorMap map[image.ID]bool, imageID image.ID, getChildren func(image.ID) []image.ID) {
+func populateImageFilterByParents(ctx context.Context, ancestorMap map[image.ID]bool, imageID image.ID, getChildren func(context.Context, image.ID) ([]image.ID, error)) error {
 	if !ancestorMap[imageID] {
-		for _, id := range getChildren(imageID) {
-			populateImageFilterByParents(ancestorMap, id, getChildren)
+		children, err := getChildren(ctx, imageID)
```

Comment:
See my other comment `getChildren()` (which is `daemon.imageService.Children`) appears to never return an error, but the changed signature appears to cause a lot of code churn to handle a possible error. Wondering why an `error` return was added to it

Reply:
github.com/containerd/containerd/images.Store.List() returns an error, so a `daemon.imageService` implemented in terms of it would need to propagate any returned error.